android/expoview/src/main/java/versioned/host/exp/exponent/VersionedUtils.kt

```diff
@@ -0,0 +1,326 @@
+          instanceManagerBuilderProperties.manifest,
+          // DO NOT EDIT THIS COMMENT - used by versioning scripts
+          // When distributing change the following two arguments to nulls
+          instanceManagerBuilderProperties.expoPackages,
+          instanceManagerBuilderProperties.exponentPackageDelegate,
+          instanceManagerBuilderProperties.singletonModules
+        )
+      )
+      .addPackage(
+        ExpoTurboPackage(
+          instanceManagerBuilderProperties.experienceProperties,
+          instanceManagerBuilderProperties.manifest
+        )
+      )
+      .setInitialLifecycleState(LifecycleState.BEFORE_CREATE)
+      .setCustomPackagerCommandHandlers(createPackagerCommandHelpers())
+      .setJavaScriptExecutorFactory(createJSExecutorFactory(instanceManagerBuilderProperties))
+    if (instanceManagerBuilderProperties.jsBundlePath != null && instanceManagerBuilderProperties.jsBundlePath!!.isNotEmpty()) {
+      builder = builder.setJSBundleFile(instanceManagerBuilderProperties.jsBundlePath)
+    }
+    return builder
+  }
+
+  private fun getDevSupportManager(reactApplicationContext: ReactApplicationContext): RNObject? {
+    val currentActivity = Exponent.instance.currentActivity
+    return if (currentActivity != null) {
+      if (currentActivity is ReactNativeActivity) {
+        currentActivity.devSupportManager
+      } else {
+        null
+      }
+    } else try {
+      val devSettingsModule = reactApplicationContext.catalystInstance.getNativeModule("DevSettings")
+      val devSupportManagerField = devSettingsModule!!.javaClass.getDeclaredField("mDevSupportManager")
+      devSupportManagerField.isAccessible = true
+      RNObject.wrap(devSupportManagerField[devSettingsModule])
+    } catch (e: Throwable) {
+      e.printStackTrace()
+      null
+    }
+  }
+
+  private fun createJSExecutorFactory(
+    instanceManagerBuilderProperties: InstanceManagerBuilderProperties
+  ): JavaScriptExecutorFactory? {
+    val appName = instanceManagerBuilderProperties.manifest.getName() ?: ""
+    val deviceName = AndroidInfoHelpers.getFriendlyDeviceName()
+
+    if (Constants.isStandaloneApp()) {
+      return JSCExecutorFactory(appName, deviceName)
+    }
+
+    val hermesBundlePair = parseHermesBundleHeader(instanceManagerBuilderProperties.jsBundlePath)
+    if (hermesBundlePair.first && hermesBundlePair.second != HERMES_BYTECODE_VERSION) {
+      val message = String.format(
+        Locale.US,
+        "Unable to load unsupported Hermes bundle.\n\tsupportedBytecodeVersion: %d\n\ttargetBytecodeVersion: %d",
+        HERMES_BYTECODE_VERSION, hermesBundlePair.second
+      )
+      KernelProvider.instance.handleError(RuntimeException(message))
+      return null
+    }
+    val jsEngineFromManifest = instanceManagerBuilderProperties.manifest.getAndroidJsEngine()
+    return if (jsEngineFromManifest != null && jsEngineFromManifest == "hermes") HermesExecutorFactory() else JSCExecutorFactory(
```

Comment:
Kotlin's == will take care of null checks:
```suggestion
    return if (jsEngineFromManifest == "hermes") HermesExecutorFactory() else JSCExecutorFactory(
```